### PR TITLE
CLEAN: fix some deprecated API usages and other compiler warnings

### PR DESCRIPTION
**Pull request by vlad20012**

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/project/toolwindow/CargoProjectsTree.kt [hunk 1]
```diff
@@ -48,7 +49,7 @@ open class CargoProjectsTree : SimpleTree() {
                 val cargoProject = selectedProject ?: return
                 // Capitalize command name to be consistent with line market providers
                 // TODO: not to use `capitalize` here
```
Comment (mati865) [inline src/main/kotlin/org/rust/cargo/project/toolwindow/CargoProjectsTree.kt:51]: Is this Todo fixed now?

Comment (vlad20012) [reply src/main/kotlin/org/rust/cargo/project/toolwindow/CargoProjectsTree.kt:51]: :thinking: @Undin ?

File: src/main/kotlin/org/rust/stdext/Utils.kt [hunk 2]
```diff
@@ -59,6 +59,8 @@ fun Path.list(): Sequence<Path> = Files.list(this).asSequence()
 
 fun String.pluralize(): String = StringUtil.pluralize(this)
 
+fun String.toCapitalized(): String = StringUtil.capitalize(this)
```
Comment (Undin) [inline src/main/kotlin/org/rust/stdext/Utils.kt:62]: Maybe `capitalized` is better?

File: src/main/kotlin/org/rust/cargo/project/toolwindow/CargoProjectsTree.kt (hunk 1 above)
Comment (Undin) [reply src/main/kotlin/org/rust/cargo/project/toolwindow/CargoProjectsTree.kt:51]: Actually, no. Let's update name of function in comment

**Review by Undin — APPROVED** [review]
Looks good in general except minor comments

**Comment by vlad20012** [comment]
bors r=Undin

**Pull request merged**